enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -772,5 +775,131 @@ describeWithToken("formatting > sandboxes", () => {
+                  type: "text",
+                },
+              },
+            },
+            type: "native",
+          },
+          display: "table",
+          visualization_settings: {},
+        }).then(({ body: { id: CARD_ID } }) => {
+          cy.log("**-- 1a. Sandbox `Orders` table based on this question --**");
+
+          cy.request("POST", "/api/mt/gtap", {
+            attribute_remappings: {
+              [ATTR_UID]: ["variable", ["template-tag", "sandbox"]],
+            },
+            card_id: CARD_ID,
+            group_id: COLLECTION_GROUP_ID,
+            table_id: ORDERS_ID,
+          });
+        });
+        cy.log("**-- 2. Create the second native question with a filter --**");
+
+        cy.request("POST", "/api/card", {
+          name: "EE_520_Q2",
+          dataset_query: {
+            database: 1,
+            native: {
+              query:
+                "SELECT * FROM PRODUCTS↵WHERE CATEGORY={{sandbox}} AND PRICE>10",
+              "template-tags": {
+                sandbox: {
+                  "display-name": "Sandbox",
+                  id: "3d69ba99-7076-2252-30bd-0bb8810ba895",
+                  name: "sandbox",
+                  type: "text",
+                },
+              },
+            },
+            type: "native",
+          },
+          display: "table",
+          visualization_settings: {},
+        }).then(({ body: { id: CARD_ID } }) => {
+          cy.log(
+            "**-- 2a. Sandbox `Products` table based on this question --**",
+          );
+
+          cy.request("POST", "/api/mt/gtap", {
+            attribute_remappings: {
+              [ATTR_CAT]: ["variable", ["template-tag", "sandbox"]],
+            },
+            card_id: CARD_ID,
+            group_id: COLLECTION_GROUP_ID,
+            table_id: PRODUCTS_ID,
+          });
+        });
+
+        cy.log("**-- 3. Fetch permissions graph --**");
+
+        cy.request("GET", "/api/permissions/graph", {}).then(
+          ({ body: { groups, revision } }) => {
+            // Update permissions for `collections` group [id: 4]
+            // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+            groups[COLLECTION_GROUP_ID] = {
+              1: {
+                schemas: {
+                  PUBLIC: {
+                    [PRODUCTS_ID]: { query: "segmented", read: "all" },
+                    [ORDERS_ID]: { query: "segmented", read: "all" },
+                  },
+                },
+              },
+            };
+
+            cy.log("**-- 4. Update/save permissions --**");
+
+            cy.request("PUT", "/api/permissions/graph", {
+              groups,
+              revision,
+            });
+          },
+        );
+
+        signOut();
+
+        cy.log("**-- Logging in as sandboxed user --**");
+        cy.request("POST", "/api/session", {
+          username: sandboxed_user.email,
+          password: sandboxed_user.password,
+        });
+      });
```

Comment:
It doesn't make any difference. `withSampleDataset()` only provides access to certain values, but doesn't affect test in any other way. I could've left every other step inside of it and it would've still worked.

But for the sake of "correctness" and easier-to-follow flow of the text, I'll definitely accept this suggestion. Thanks.